libs/lib-client/diagram/src/lib/diagram/diagram-row.component.ts

```diff
@@ -76,4 +77,8 @@ export class DiagramRowComponent implements OnChanges {
       this.action.emit(actionEventFactory.move(itemId, parentId));
     }
   }
+
+  restrictTileDrop = (dragEvent: CdkDrag) => {
+    return !this.rowParents?.length || !this.rowParents.includes(dragEvent.data);
```

Comment:
Should it not be `&&` ?
```suggestion
    return !this.rowParents?.length && !this.rowParents.includes(dragEvent.data);
    /* May be giving the responsibility to the service gives advantage of re-usability 
    something like this.dragService.isAllowedToEnterRow(dragEvent.data) */
```

Reply:
No it should be || only.
Condition is to see if row has any parents and then see if it has drag item as its parent.

Reply:
Agree with re-usability, move logic to service. Thanks @apandurangi